proto/connect/federation_service/v1alpha1/federation_service.proto

```diff
@@ -0,0 +1,50 @@
+// Copyright 2025 Cofide Limited.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package proto.connect.federation_service.v1alpha1;
+
+import "proto/federation/v1alpha1/federation.proto";
+
+option go_package = "github.com/cofide/cofide-api-sdk/gen/go/proto/connect/federation_service/v1alpha1";
+
+service FederationService {
+  rpc CreateFederation(CreateFederationRequest) returns (CreateFederationResponse) {}
+  rpc ListFederations(ListFederationsRequest) returns (ListFederationsResponse) {}
+  rpc DeleteFederation(DeleteFederationRequest) returns (DeleteFederationResponse) {}
```

Comment:
I went with destroy in the cluster API, and it's also used in a place or two in cofidectl. Both seem reasonable but we should be consistent.